pgjdbc/src/main/java/org/postgresql/jdbc/PgConnection.java

```diff
@@ -322,8 +322,7 @@ public PgConnection(HostSpec[] hostSpecs,
 
     fieldMetadataCache = new LruCache<FieldMetadata.Key, FieldMetadata>(
             Math.max(0, PGProperty.DATABASE_METADATA_CACHE_FIELDS.getInt(info)),
-            Math.max(0, PGProperty.DATABASE_METADATA_CACHE_FIELDS_MIB.getInt(info) * 1024 * 1024),
-        false);
+            Math.max(0, PGProperty.DATABASE_METADATA_CACHE_FIELDS_MIB.getInt(info) * 1024L * 1024L));
```

Comment:
These should be in a separate commit, and possibly includes other sources files with the same problem. The essence of it seems to be an overflow concern, where adding "L" to the end, implies "long" calculations instead of the prior "int". It might be technically a problem for overflow, but it has nothing to do with "fix: LruCache evict on usage rather than insertion".